src/main/java/org/terasology/minecarts/controllers/CartMotionSystem.java

```diff
@@ -172,12 +173,24 @@ private void updateCart(EntityRef railVehicle, float delta) {
                 Util.bound(railVehicleComponent.velocity);
                 if (pathFollowerSystem.move(railVehicle, Math.signum(segmentVehicleComponent.heading.dot(railVehicleComponent.velocity)) * mag * delta, segmentMapping)) {
 
-                    //calculate the cart rotation
-                    Quat4f horizontalRotation = Quat4f.shortestArcQuat(Vector3f.north(), new Vector3f(segmentVehicleComponent.heading).setY(0).normalize());
-                    Quat4f verticalRotation = Quat4f.shortestArcQuat(new Vector3f(segmentVehicleComponent.heading).setY(0).normalize(), new Vector3f(segmentVehicleComponent.heading));
-                    verticalRotation.mul(horizontalRotation);
-                    location.setLocalRotation(verticalRotation);
-                    location.setWorldPosition(position);
+                    Vector3f frontAxisPosition = pathFollowerSystem.vehiclePoint(railVehicle, railVehicleComponent.frontAxisOffset, segmentMapping);
+                    Vector3f backAxisPosition = pathFollowerSystem.vehiclePoint(railVehicle, railVehicleComponent.backAxisOffset, segmentMapping);
+                    if (frontAxisPosition == null || backAxisPosition == null) {
+                        Quat4f horizontalRotation = Quat4f.shortestArcQuat(Vector3f.north(), new Vector3f(segmentVehicleComponent.heading).setY(0).normalize());
+                        Quat4f verticalRotation = Quat4f.shortestArcQuat(new Vector3f(segmentVehicleComponent.heading).setY(0).normalize(), new Vector3f(segmentVehicleComponent.heading));
+                        verticalRotation.mul(horizontalRotation);
+                        location.setLocalRotation(verticalRotation);
+                        location.setWorldPosition(position);
+                    } else {
+                        frontAxisPosition.y = mesh.mesh.getAABB().getMax().y / 2.0f + frontAxisPosition.y + .01f;
+                        backAxisPosition.y = mesh.mesh.getAABB().getMax().y / 2.0f + backAxisPosition.y + .01f;
```

Comment:
I think it would be better making `.01f` a constant with a name that clarifies its function. Is it an offset of some kind, and why is it used?

Reply:
not sure what the offset is for.